### DISCUSSION TITLE
In the SceneParameters, how to know one's parent?

### DISCUSSION THREAD

**Discussion by Mephisto405** [mitsuba-renderer/mitsuba3#485] [Q&A] [answered]
For instance, when we load a scene file, we can see a table of SceneParameters with columns `Name`, `Flags`, `Type`, and `Parent` as below.
What I want to do is to get the parent of an instance.
That is, I want to make sure the parent of the `params['Floor.to_world']` is `Rectangle`.
Ultimately, I want to obtain the world transformation of shapes whose parents are `Rectangle` to get the coordinates of the four corner points of all rectangles.
```
scene = mi.load_file("data/cornell-box/scene_v3.xml")
params = mi.traverse(scene)
print(params)
```
![image](https://user-images.githubusercontent.com/25499931/211479662-9bcaaaf4-faa2-494e-84ce-83199ee37859.png)

**Comment by njroussel** [comment]
Hi @Mephisto405 

I don't believe there's a built-in mechanism to do this. You can either parse the output of `mi.traverse()` to detect parameters which have the `Parent` you are searching for, or you can rename your scene objects in your scene description to include this information and then use [SceneParameters.keep(some_regex_that_matches_your_parent_type)](https://mitsuba.readthedocs.io/en/stable/src/api_reference.html#mitsuba.SceneParameters.keep)

**Comment by ziyi-zhang** [comment, marked as answer]
> I want to obtain the world transformation of shapes whose parents are Rectangle to get the coordinates of the four corner points of all rectangles

If it's for visualization purpose, I have the following workaround
```
# iterate over all shapes
obj = scene.shapes()[idx]
params = mi.traverse(obj)
if obj.__repr__().startswith('Rectangle'):
        v = np.array([[-1.0, -1, 0], [1, -1, 0], [1, 1, 0], [-1, 1, 0]])
        f = np.array([[0, 1, 2], [0, 2, 3]])
        v_ = params["to_world"] @ mi.Point3f(v)
        v = v_.numpy()
```

**Comment by Mephisto405** [comment]
@njroussel @ziyi-zhang 
Yo, early bird bros!!! So thankful for your fast reply and helps XD. 
I now feel safe that many guys would help me if I do something wrong with Mitsuba 3. kkk
Very thankful again! 

I just found another mechanism to achieve this, so I am sharing it here:
```
# it use `properties` dictionary of `SceneParameters` object
params.properties['Floor.to_world'][2].__repr__().startswith('Rectangle')
```
